packages/walletconnect/src/index.ts

```diff
@@ -173,9 +173,18 @@ export class WalletConnect extends Connector {
           if (error?.message === 'User closed modal') await this.deactivate()
           throw error
         })
-      const chainId = await this.provider.request<string>({ method: 'eth_chainId' })
-
-      this.actions.update({ chainId: parseChainId(chainId), accounts })
+      const chainId = parseChainId(await this.provider.request<string>({ method: 'eth_chainId' }))
+      /**
+       * It is possible that the user has changed the chain in the wallet while the modal was open.
+       * In that case, the `desiredChainId` will differ from the `chainId` returned by the wallet,
+       * resulting in incorrect `rpc` endpoint being used. Unfortunately, there's no public API
+       * to set the `rpc` endpoint, so we have to use the private API instead.
+       */
+      if (chainId !== desiredChainId) {
+        // @ts-ignore
```

Comment:
for ts-ignore i usually like to specify the rule im ignoring~ (so it will still fail if theres a new ts issue in the future)

Reply:
You should also specify why you're ignoring it and, ideally, include a TODO with a jira ticket linked (eg `TODO(INFRA-123): This is the reason I need ts-ignore.`).

Reply:
@tinaszheng are you sure you can ignore a specific issue? https://github.com/microsoft/TypeScript/issues/19139